src/main/ui/conversation-control.gd

```diff
@@ -1,10 +1,101 @@
 extends Control
 """
 Displays UI elements for having a conversation.
+
+The chat window is decorated with objects in the background called 'accents'. These accents can be injected with the
+set_accent_def function to configure the chat window's appearance.
 """
 
+const PULSE_PERIOD = 5.385
+const PULSE_AMOUNT = Vector2(0.006, 0.012)
+
+onready var accent_textures := [
+	preload("res://assets/ui/chat/texture/bg00.png"),
+	preload("res://assets/ui/chat/texture/bg01.png"),
+	preload("res://assets/ui/chat/texture/bg02.png"),
+	preload("res://assets/ui/chat/texture/bg03.png"),
+	preload("res://assets/ui/chat/texture/bg04.png"),
+	preload("res://assets/ui/chat/texture/bg06.png"),
+	preload("res://assets/ui/chat/texture/bg07.png"),
+	preload("res://assets/ui/chat/texture/bg08.png"),
+	preload("res://assets/ui/chat/texture/bg09.png"),
+	preload("res://assets/ui/chat/texture/bg10.png"),
+	preload("res://assets/ui/chat/texture/bg11.png"),
+	preload("res://assets/ui/chat/texture/bg12.png"),
+	preload("res://assets/ui/chat/texture/bg13.png"),
+	preload("res://assets/ui/chat/texture/bg14.png"),
+	preload("res://assets/ui/chat/texture/bg15.png")
+]
+
+var _total_time := 0.0
+
 """
 Animates the conversation UI to gradually reveal the specified text, mimicking speech.
 """
-func play_text(text_with_pauses: String) -> void:
-	$Holder/Panel/Label.play_text(text_with_pauses)
+func play_text(text_with_pauses: String, accent_def: Dictionary) -> void:
+	set_accent_def(accent_def)
+	$Holder/ChatLabel.play_text(text_with_pauses)
+
+
+"""
+Configures the conversation window's appearance based on the 'accent_def' which is passed in.
+
+The conversation window changes its appearance based on who is talking. For example, one character's speech might be
+blue with a black background, and giant blue soccer balls in the background.
+
+The 'accent_def' passed in includes details about the conversation window, such as 'blue', 'soccer balls' and
+'giant'.
+
+Parameters:
+	'accent_def': Describes the colors and textures used to draw the conversation window.
+"""
+func set_accent_def(accent_def: Dictionary) -> void:
+	set_accent_texture(accent_def["accent_texture"] if accent_def.has("accent_texture") else 0)
+	set_accent_scale(accent_def["accent_scale"] if accent_def.has("accent_scale") else 2.0)
+	set_accent_inverted(accent_def["accent_inverted"] if accent_def.has("accent_inverted") else false)
+	set_color(accent_def["color"] if accent_def.has("color") else Color.gray,
+		accent_def["dark"] if accent_def.has("dark") else false)
+
+
+func set_accent_inverted(inverted: bool) -> void:
```

Comment:
These setters can be private; nobody calls them.

Reply:
Additionally, these setters might be better as something like `apply_accent_inverted`, `apply_accent_texture`, `apply_accent_scale` and they can take a single `accent_def` parameter.

Reply:
Refactored into `_apply_accent_colors()`, `_apply_accent_shapes()`.